surefire-providers/surefire-junit4/src/main/java/org/apache/maven/surefire/junit4/JUnit4Provider.java

```diff
@@ -55,7 +53,8 @@
 import static java.lang.reflect.Modifier.isAbstract;
 import static java.lang.reflect.Modifier.isInterface;
 import static org.apache.maven.surefire.booter.CommandReader.getReader;
-import static org.apache.maven.surefire.common.junit4.JUnit4ProviderUtil.generateFailingTests;
+import static org.apache.maven.surefire.common.junit4.FilterFactory.createMatchDescriptionsFilter;
```

Comment:
It seems we can move this method to `JUnit4ProviderUtil` and remove `org.apache.maven.surefire.common.junit4.FilterFactory`.